gr-azure-software-radio/python/blob_source.py

```diff
@@ -176,29 +186,32 @@ def work(self, _, output_items):
             self.blob_iter = self.setup_blob_iterator()
             self.first_run = False
 
-        out = output_items[0]
-        noutput_items = len(out)
+        # get a view into the output buffer as a 1D array of bytes, so our code can copy over the data
+        # without caring about the actual datatype or vlen in use
+        out = output_items[0].reshape((output_items[0].size,)).view(dtype=np.byte)
```

Comment:
how much overhead does reshape add here? 

I guess the though is this has less overhead then like converting the raw bytes with like np.frombuffer or something along those lines?

Reply:
Reshapes are supposed to return a view into the array without doing a copy, as long as the array is contiguous. If we want to be absolutely sure I guess we could split this up a bit, use only view objects, and just update the shape of the view object which throws an error if it can't be done. 

The idea here is that both the reshape and the view are only updating numpy metadata about the underlying memory, not actually moving data around.